src/dispatch/plugins/dispatch_slack/case/messages.py

```diff
@@ -298,6 +305,116 @@ def create_signal_messages(case_id: int, channel_id: str, db_session: Session) -
     return Message(blocks=signal_metadata_blocks).build()["blocks"]
 
 
+def create_genai_signal_summary(
+    case: Case,
+    channel_id: str,
+    db_session: Session,
+    client: WebClient,
+) -> list[Block]:
+    """Creates enhanced signal messages with historical context."""
+    signal_metadata_blocks: list[Block] = []
+
+    signal_instances_query = (
+        db_session.query(SignalInstance, Signal)
+        .join(Signal)
+        .with_entities(SignalInstance.id, Signal)
+        .filter(SignalInstance.case_id == case.id)
+        .order_by(SignalInstance.created_at)
+    )
+
+    (first_instance_id, first_instance_signal) = signal_instances_query.first()
+
+    # Fetch up to 10 recent related cases
+    related_cases = (
+        db_session.query(Case)
+        .join(SignalInstance)
+        .filter(SignalInstance.signal_id == first_instance_signal.id)
+        .filter(Case.id != case.id)  # Exclude the current case
+        .order_by(desc(Case.created_at))
+        .limit(10)
+        .all()
+    )
+
+    # Prepare historical context
+    historical_context = []
+    for related_case in related_cases:
+        historical_context.append(f"Case: {related_case.name}")
+        historical_context.append(f"Resolution: {related_case.resolution}")
+        historical_context.append(f"Resolution Reason: {related_case.resolution_reason}")
+
+        # Fetch Slack messages for the related case
+        if related_case.conversation and related_case.conversation.channel_id:
+            try:
+                # Fetch threaded messages
+                thread_messages = client.conversations_replies(
+                    channel=related_case.conversation.channel_id,
+                    ts=related_case.conversation.thread_id,
+                )
+
+                # Add relevant messages to the context (e.g., first 5 messages)
+                for message in thread_messages["messages"][:5]:
+                    historical_context.append(f"Slack Message: {message['text']}")
+            except SlackApiError as e:
+                log.error(f"Error fetching Slack messages for case {related_case.name}: {e}")
+
+        historical_context.append("---")
+
+    historical_context_str = "\n".join(historical_context)
+
+    signal_instance = signal_service.get_signal_instance(
+        db_session=db_session, signal_instance_id=first_instance_id
+    )
+
+    genai_plugin = plugin_service.get_active_instance(
+        db_session=db_session, project_id=case.project.id, plugin_type="artificial-intelligence"
+    )
+    if not genai_plugin:
+        return signal_metadata_blocks
+
+    response = genai_plugin.instance.chat_completion(
+        prompt=f"""
```

Comment:
You should be able to fetch the prompt with the exception of current case, ... from the signal definition with the changes I made.